numba/core/config.py

```diff
@@ -108,6 +108,20 @@ def update(self, force=False):
             # Store a copy
             self.old_environ = dict(new_environ)
 
+        self.validate()
+
+    def validate(self):
+        if CUDA_USE_CUDA_PYTHON:  # noqa: F821
+            try:
+                import cuda  # noqa: F401
+            except ImportError as ie:
+                msg = ("CUDA Python bindings requested, "
+                       "but they are not importable")
+                raise RuntimeError(msg) from ie
+
+            if CUDA_PER_THREAD_DEFAULT_STREAM:  # noqa: F821
+                warnings.warn("PTDS is not supported with CUDA Python")
```

Comment:
xref https://github.com/numba/numba/pull/7461#issuecomment-958925392 not supported yet. What happens if someone tries using `PTDS` with the CUDA python package, they get this warning.... and then what?! Does the program carry on and work "correctly" but without `PTDS`?

Reply:
Yes - this won't affect correctness, it will just have some impact on performance in some cases. So I'd rather things run a little slower with a warning, than not run at all.

Reply:
Agree, thanks for confirming.